api/src/controllers/users.ts

```diff
@@ -142,4 +147,92 @@ export default class UsersController {
 
         return res.status(204).json({ msg: "Logged out successfully." });
     }
+
+    public async getFriends(req: Request, res: Response) {
+        const userRepository = this.repository;
+        const friendshipRepository = this.friendshipRepository;
+
+        const userId = req.params.id;
+
+        const friendships = await friendshipRepository.find({ user_id1: userId });
+
+        if (friendships.length === 0) {
+            return res.status(200).json([]);
+        }
+
+        const friends = await userRepository.findByIds(friendships.map(friendship => friendship.user_id2));
+        return res.status(200).json(friends);
+    }
+
+    public async acceptFriend(req: Request, res: Response) {
+        const friendshipRepository = this.friendshipRepository;
+
+        const user1Id = Number(req.params.id);
+        const user2Id = Number(req.body.friendId);
+
+        const friendship = await friendshipRepository.findOne({ where: { user_id1: user1Id, user_id2: user2Id } });
+
+        if (friendship.status !== FriendshipStatus.PENDING) {
+            throw new Error('Friendship not pending');
+        }
+
+        friendship.status = FriendshipStatus.ACCEPTED;
+        await friendshipRepository.save(friendship);
+    }
+
+    public async addFriend(req: Request, res: Response) {
+        const userRepository = this.repository;
+        const friendshipRepository = this.friendshipRepository;
+
+        const user1Id = Number(req.params.id);
+        const friendId = Number(req.body.friendId);
+        // Check if users exist
+        const user1 = await userRepository.find({
+            where: {
+                Id: user1Id
+            }
+        });
+
+        const user2 = await userRepository.find({
+            where: {
+                Id: friendId
+            }
+        });
+
+        if (!user1 || !user2) {
+            throw new Error('One or both users not found');
+        }
+
+        // Check if friendship already exists
+        const existingFriendship = await friendshipRepository.findOne({ where: { user_id1: user1Id, user_id2: friendId } });
+
+        if (existingFriendship) {
+            throw new Error('Friendship already exists');
+        }
+
+        // Create new friendship
+        const newFriendship = new Friendship();
+        newFriendship.user_id1 = user1Id;
+        newFriendship.user_id2 = friendId;
+        newFriendship.status = FriendshipStatus.PENDING;
+        newFriendship.action_user_id = user1Id; // The user who initiated the friendship is the action_user
+
+        // Save the friendship
+        await friendshipRepository.save(newFriendship);
+        return res.status(200).json(newFriendship);
+    }
+
+    public async getfriends(req: Request, res: Response) {
+        const friendshipRepository = this.friendshipRepository;
```

Comment:
This is returning the full DB object, we should remove fields that we don't want to expose (ie: `password` and some others)